homeassistant/components/sensor/zamg.py

```diff
@@ -62,19 +60,31 @@
 PLATFORM_SCHEMA = cv.PLATFORM_SCHEMA.extend({
     vol.Required(CONF_MONITORED_CONDITIONS):
         vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)]),
-    vol.Required(CONF_STATION_ID):
-        vol.All(cv.string, vol.In(VALID_STATION_IDS)),
+    vol.Optional(CONF_STATION_ID): cv.string,
```

Comment:
According to the `Stationliste` the IDs have no leading zeros. Most IDs are 11xxx and one exception (8989042) as I'm just looking at the public available information.